x-pack/plugins/apm/public/components/alerting/ui_components/apm_rule_group_by.tsx

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiComboBox } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import React, { useCallback } from 'react';
+import { TRANSACTION_NAME } from '../../../../common/es_fields/apm';
+
+interface Props {
+  options: { groupBy: string[] | string | undefined };
+  preSelectedOptions: string[];
+  onChange: (groupBy: string | null | string[]) => void;
+  errorOptions?: string[];
+}
+
+const fields: string[] = [TRANSACTION_NAME];
```

Comment:
I expect the fields (options of the combo box) to be defined by the consumer. Similar to `preSelectedOptions`. Currently, all the rules that use the component will have the `transaction.name` option